src/db/spanner/manager/session.rs

```diff
@@ -63,7 +67,23 @@ pub async fn create_spanner_session(
 pub async fn recycle_spanner_session(
     conn: &mut SpannerSession,
     database_name: &str,
+    max_lifetime: Option<u32>,
 ) -> Result<(), DbError> {
+    if let Some(max_life) = max_lifetime {
```

Comment:
Adding a metric for both expiring cases might be handy. The Manager already has a `metrics` instance.

Reply:
oh, cool! yeah, will do.